buffer/src/main/java/io/netty/buffer/api/BufferAllocator.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api;
+
+import io.netty.buffer.api.pool.PooledBufferAllocator;
+
+import java.nio.ByteOrder;
+import java.util.function.Supplier;
+
+/**
+ * Interface for allocating {@link Buffer}s.
+ */
+public interface BufferAllocator extends AutoCloseable {
+    /**
+     * Produces a {@link BufferAllocator} that allocates unpooled, on-heap buffers.
+     * On-heap buffers have a {@code byte[]} internally, and their {@linkplain Buffer#nativeAddress() native address}
+     * is zero.
+     * <p>
+     * The concrete {@link Buffer} implementation is chosen by {@link MemoryManager#instance()}.
+     *
+     * @return A non-pooling allocator of on-heap buffers
+     */
+    static BufferAllocator heap() {
+        return new ManagedBufferAllocator(MemoryManager.instance(), false);
+    }
+
+    /**
+     * Produces a {@link BufferAllocator} that allocates unpooled, off-heap buffers.
+     * Off-heap buffers a native memory pointer internally, which can be obtained from their
+     * {@linkplain Buffer#nativeAddress() native address method.
+     * <p>
+     * The concrete {@link Buffer} implementation is chosen by {@link MemoryManager#instance()}.
+     *
+     * @return A non-pooling allocator of on-heap buffers
+     */
+    static BufferAllocator direct() {
+        return new ManagedBufferAllocator(MemoryManager.instance(), true);
+    }
+
+    /**
+     * Produces a pooling {@link BufferAllocator} that allocates and recycles on-heap buffers.
+     * On-heap buffers have a {@code byte[]} internally, and their {@linkplain Buffer#nativeAddress() native address}
+     * is zero.
+     * <p>
+     * The concrete {@link Buffer} implementation is chosen by {@link MemoryManager#instance()}.
+     *
+     * @return A pooling allocator of on-heap buffers
+     */
+    static BufferAllocator pooledHeap() {
```

Comment:
out of curiosity: why use the longer "pooled" here, instead of making the other one "unpooled" ?   Is pooled memory being phased out?

Reply:
No. I've aligned the names.